pkg/validation/spec/header.go

```diff
@@ -92,14 +92,14 @@ func (h *Header) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Dec
 		return err
 	}
 
+	internal.SanitizeExtensions(x.Extensions)
+	if len(x.Extensions) == 0 {
+		x.Extensions = nil
+	}
```

Comment:
Extensions is aliased to `map[string]interface{}` which can be passed into sanitize because the types match. It seems like a pointer `*Extensions` is different from `*map[string]interface{}` and cannot be used directly. Are we okay with casting before the sanitize? I'm not exactly sure how much this would improve readability

Reply:
Yeah, that's not great, let's keep it as it is, thanks!